config/config.php

```diff
@@ -112,6 +112,25 @@
     'blacklist_grace_period' => env('JWT_BLACKLIST_GRACE_PERIOD', 0),
 
     /*
+    | -------------------------------------------------------------------------
+    | Retrieve user on token authentication
+    | -------------------------------------------------------------------------
+    |
+    | After a token has been issued the user id might no longer exists
+    | what would let the token out of sync with the database.
+    | To ensure that the token is not only valid but also refers to an
+    | existing user the user could be automatically retrieved from the database
+    | every time a token is validated.
+    |
+    | Setting it to "true" will retrieve the user every time the token is checked.
+    | Setting it to "false will query the database only when Auth::user() is invoked
+    | for the first time.
+    |
+    */
+
+    'retrieve_user_on_authentication' => true,
```

Comment:
I'm afraid |'m still not sold on adding this to the global config, as it is quite a unique requirement for most people I would say. And I don't want to add too much in here, as there are already lots of options.

I would class it more as advanced config and that has it's place in a custom service provider in my eyes.

Something like:

```php
// In your AppServiceProvider or something
app('tymon.jwt.auth')->setRetrieveUserOnAuthentication(true);
```

Thoughts?